### PR TITLE
Add support for sync UI on Android

### PR DESCRIPTION
**Pull request by emerick**
Fix https://github.com/brave/brave-browser/issues/4833

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: build/android/config.gni [hunk 1]
```diff
@@ -5,4 +5,10 @@ declare_args() {
 }
 
 brave_android_manifest_includes = [ "//brave/android/java/AndroidManifest.xml" ]
-brave_chrome_java_deps = ["//brave/third_party/android_deps:google_play_services_safetynet_java"]
+
+brave_chrome_java_deps = [
+  "//brave/third_party/android_deps:google_play_services_safetynet_java",
+  "//brave/third_party/zxing:zxing_java",
```
Comment (bridiver) [inline build/android/config.gni]: what is this stuff and is any of it actually relevant to the UI or was this stuff for the sync backend?

Comment (bridiver) [reply build/android/config.gni]: I see, it's for QR code reading?

File: android/java/org/chromium/chrome/browser/BraveApplication.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+package org.chromium.chrome.browser;
+
+import android.app.Application;
+
+import org.chromium.chrome.browser.BraveSyncWorker;
+
+public class BraveApplication extends Application {
+    // Sync worker
+    public BraveSyncWorker mBraveSyncWorker;
```
Comment (bridiver) [inline android/java/org/chromium/chrome/browser/BraveApplication.java]: this class will never be used in brave-core so we should just leave it out

File: build/android/config.gni (hunk 1 above)
Comment (emerick) [reply build/android/config.gni]: Yes, it's the QR code libraries.

File: android/java/org/chromium/chrome/browser/BraveApplication.java (hunk 2 above)
Comment (emerick) [reply android/java/org/chromium/chrome/browser/BraveApplication.java]: Removed the `ChromeApplication` patch and replaced `BraveSyncWorker` with `BraveSyncService` as recommended. `BraveSyncService` doesn't yet implement the actual sync service API as the goal here was mainly to get the UI up and running with the understanding that the actual service will be implemented later.

I also fixed the preferences in `BraveSyncScreensPreference` to refer to the appropriate brave-core preferences.

**Comment by emerick** [comment]
Lint failed in CI due to unrelated `import httplib2` error and previous CI run passed. Going to go ahead and merge as the Java lint occurs as part of the build process and it's running successfully on my local checkout.

**Pull request merged**